contracts/test/GnosisSafeABICreator.sol

```diff
@@ -3,5 +3,6 @@ pragma solidity ^0.8.15;
 
 // Getting ABIs for the Gnosis Safe master copy and proxy contracts
 import "@gnosis.pm/safe-contracts/contracts/GnosisSafe.sol";
+import "@gnosis.pm/safe-contracts/contracts/handler/DefaultCallbackHandler.sol";
```

Comment:
DefaultCallbackHandler is needed since the ERC721 support is required for the service owner multisig.

Reply:
@kupermind make sure that the fronend makes this known to the user. Maybe we can add a check and error message if it’s not set.

Reply:
All gnosis safe contracts created via Gnosis Safe App have at least the `DefaultCallbackHandler` by default. Those of the latest version 1.3 have the `CompatibilityFallbackHandler` by default. The only way to check that is to read the storage of the proxy contract.

We can have the message when we identify that it's a safe that wants to create a component / agent / service, also we can have an additional check for the proxy storage fallback handler slot.

Reply:
Then get the address by a slot:
`FALLBACK_HANDLER_STORAGE_SLOT = 0x6c9a6c4a39284e37ed1cf53d337577d14212a4870fb976a4366c693b939918d5`

Then read the value of the pure `onERC721Received()` function

Reply:
python:
`slot = w3_mainnet.eth.getStorageAt(gnosis_safe_address_reference,FALLBACK_HANDLER_STORAGE_SLOT).hex()`

Reply:
Suggest you open an issue for Mohan

Reply:
It's done already in 2 tickets )